test/test_torch.py

```diff
@@ -341,14 +341,33 @@ def test_csub(self):
         res_csub.sub_(scalar)
         self.assertEqual(res_add, res_csub)
 
-    def test_neg(self):
-        a = torch.randn(100, 90)
-        zeros = torch.Tensor().resize_as_(a).zero_()
+    @staticmethod
+    def _test_neg(self, cast):
+        float_types = ['torch.DoubleTensor', 'torch.FloatTensor', 'torch.LongTensor']
+        int_types = ['torch.IntTensor', 'torch.ShortTensor']
+
+        for t in float_types + int_types:
```

Comment:
that will only give integer values for floating point types, right?  Given all the template/macro expansion it seemed good to check we preserve floats.